src/MongoDbEventStoreAdapter.php

```diff
@@ -254,17 +268,15 @@ public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata
 
             $createdAt = new \DateTime();
             $createdAt->setTimestamp($eventData['created_at']->sec);
-
-            $events[] = $eventClass::fromArray(
-                [
-                    'uuid' => $eventData['_id'],
-                    'name' => $eventData['event_name'],
-                    'version' => (int) $eventData['version'],
-                    'created_at' => $createdAt->format(\DateTime::ISO8601),
-                    'payload' => $eventData['payload'],
-                    'metadata' => $metadata
-                ]
-            );
+            $createdAt->setTimezone(new \DateTimeZone('UTC'));
+
+            $events[] = $this->messageFactory->createMessageFromArray($eventData['event_name'], [
+                'uuid' => $eventData['_id'],
+                'version' => (int) $eventData['version'],
+                'created_at' => $createdAt->format(\DateTime::ISO8601),
```

Comment:
Can we store `created_at` as a string instead of a \MongoDate? 
One decision I want to move to the user is: In which format should the date be stored.
With a custom `MessageConverter` a user can for example include microseconds in the formatted date and should get this format back in a custom `MessageFactory`.

Or we align the interfaces again and say that a date MUST be given or formatted as ISO8601 or as ISO8601 + microseconds.

What do you think?